### ISSUE TITLE
Problem with relative step sizes

### ISSUE THREAD

**Issue by jaredcsiegel** [pmelchior/scarlet2#18]
Using either
`from scarlet2 import relative_step\
Parameter(jnp.ones(5), constraint=constraints.positive, stepsize=relative_step)`
or 
`Parameter(jnp.ones(5), constraint=constraints.positive, stepsize=lambda x: jnp.linalg.norm(x)*1e-1)`
is yielding nans in the loss function and final models that are all nan.

**Comment by pmelchior** [comment]
I sent these as examples how to set up parameters, they still need to make sense in your case, e.g. does all-ones for this parameter make sense _in your case_?

Also, the second version is not recommended because `constraints.positive` is a log transform, applied to a vector of ones gives a norm of 0, so step size is 0. This parameter is frozen.

**Comment by jaredcsiegel** [comment]
Of course, for my case I simply replaced `stepsize=1e-1 `(or something similar) with `stepwise = relative_step`. Upon inspection, this works well for the spectrum initialization:

**Comment by jaredcsiegel** [comment]
The problem was ultimately with initialization of the parameters (the initial step size was considerably larger than intended and led to the bad fits, lowering `factor` in `relative_step ` solved the issue).
